website/content/docs/job-specification/disconnect.mdx

```diff
@@ -0,0 +1,202 @@
+  shorter than ttl, and `replace` = false, can lead to a state where no allocations
+  are running, hence it is not admitted and will cause a validation error.
+
+  The Nomad client process must be running for this to occur. This setting
+  cannot be used with [`lost_after`].
+
+- `reconcile` `(string: “best_score”)` - Specifies which allocation to keep once
+  the previously disconnected node regains connectivity.
+  It has four possible values which are described below:
+
+    - `keep_original`: Always keep the original allocation, bare in mind 
+    when choosing this option, it can have crashed while the client was 
+    disconnected.
+    - `keep_replacement`: Always keep the allocation that was rescheduled 
+    to replace the disconnected one.
+    - `best_score`: Keep the allocation running on the node with the best 
+    score.
+    - `longest_running`: Keep the allocation that has been up and running 
+    continuously for the longest time.
+
+
+## `disconnect` Examples
+
+The following examples only show the `disconnect` blocks. Remember that the
+`disconnect` block is only valid in the placements listed above.
+
+### Stop After
+
+This example shows how `stop_after` interacts with
+other blocks. For the `first` group, after the default 10 second
+[`heartbeat_grace`] window expires and 90 more seconds passes, the
+server will reschedule the allocation. The client will wait 90 seconds
+before sending a stop signal (`SIGTERM`) to the `first-task`
+task. After 15 more seconds because of the task's `kill_timeout`, the
+client will send `SIGKILL`. The `second` group does not have
+`stop_after`, so the server will reschedule the
+allocation after the 10 second [`heartbeat_grace`] expires. It will
+not be stopped on the client, regardless of how long the client is out
+of touch.
+
+Note that if the server's clocks are not closely synchronized with
+each other, the server may reschedule the group before the client has
+stopped the allocation. Operators should ensure that clock drift
+between servers is as small as possible.
+
+Note also that a group using this feature will be stopped on the
+client if the Nomad server cluster fails, since the client will be
+unable to contact any server in that case. Groups opting in to this
+feature are therefore exposed to an additional runtime dependency and
+potential point of failure.
+
+```hcl
+group "first" {
+  stop_after_client_disconnect = "90s"
+
+  task "first-task" {
+    kill_timeout = "15s"
+  }
+}
+
+group "second" {
+
+  task "second-task" {
+    kill_timeout = "5s"
+  }
+}
+```
+
+### Lost After
+
+`lost_after` specifies a duration during which a Nomad client will
+attempt to reconnect allocations after it fails to heartbeat in the
+[`heartbeat_grace`] window.
+
+By default, allocations running on a client that fails to heartbeat will be
+marked "lost". When a client reconnects, its allocations, which may still be
+healthy, will restart because they have been marked "lost". This can cause
```

Comment:
Will they restart or will they have been rescheduled, so they stop?

Also, I would consider moving this entire paragraph somewhere near the top of this page, so that we have a clear description of the default behavior before describing how `disconnect` changes it.